src/status_im/chat/handlers/commands.cljs

```diff
@@ -38,11 +40,14 @@
                 params   {:parameters params
                           :context    (generate-context db chat-id to)}
                 callback #(let [result (get-in % [:result :returned])
-                                result (if (:markup result)
+                                result' (if (:markup result)
                                          (update result :markup cu/generate-hiccup)
-                                         result)]
-                            (dispatch [:set-in [:message-data data-type message-id] result])
-                            (when on-requested (on-requested result)))]
+                                         result)] 
+                            (dispatch [:set-in [:message-data data-type message-id] result'])
+                            (when (= :preview data-type)
```

Comment:
The problem with this approach is that `:markup` in `result` is already updated with `cu/generate-hiccup` (which does translations from keyword to native rn components) before being passed as argument to `on-requested` (if defined), we don't want that as we need to store markup serialised into string in realm db.

What do you think about not doing the `cu/generate-hiccup` translation at write time at all, rather offloading this to subscriptions. 
It will simplify thing little bit and provide better inspect/debug view over app-db, as we will be storing markup there as simple vectors with component keywords (eq. `[:text {} "blah"]`) instead of being stored already translated to rn components.

In the same time, it would be probably good to already break-up the `:request-command-data` handler into pure handler with associated effects/coeffects, akin to work @flexsurfer is currently doing for contacts, do you agree ?